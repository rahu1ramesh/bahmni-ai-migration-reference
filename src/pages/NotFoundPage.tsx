import React from 'react';
import { Grid, Column, Heading, Section, Link } from '@carbon/react';
import { Link as RouterLink } from 'react-router-dom';

const NotFoundPage: React.FC = () => {
  return (
    <Section>
      <Grid>
        <Column lg={16} md={8} sm={4}>
          <Heading>404 - Page Not Found</Heading>
          <p>The page you are looking for does not exist.</p>
          <Link as={RouterLink} to="/">
            Return to Home
          </Link>
        </Column>
      </Grid>
    </Section>
  );
};

export default NotFoundPage;
