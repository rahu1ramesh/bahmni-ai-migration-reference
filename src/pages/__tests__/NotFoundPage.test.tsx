import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NotFoundPage from '../NotFoundPage';

describe('NotFoundPage Component', () => {
  const renderWithRouter = (ui: React.ReactElement) => {
    return render(ui, { wrapper: BrowserRouter });
  };

  it('should render without crashing', () => {
    renderWithRouter(<NotFoundPage />);
    expect(screen.getByText('404 - Page Not Found')).toBeInTheDocument();
  });

  it('should display the error message', () => {
    renderWithRouter(<NotFoundPage />);
    expect(
      screen.getByText('The page you are looking for does not exist.'),
    ).toBeInTheDocument();
  });

  it('should render a link back to home page', () => {
    renderWithRouter(<NotFoundPage />);
    const homeLink = screen.getByText('Return to Home');
    expect(homeLink).toBeInTheDocument();
    expect(homeLink.closest('a')).toHaveAttribute('href', '/');
  });

  it('should render with correct structure', () => {
    renderWithRouter(<NotFoundPage />);
    // Test for the presence of the heading and paragraph
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(
      screen.getByText('The page you are looking for does not exist.'),
    ).toBeInTheDocument();
    expect(screen.getByText('Return to Home')).toBeInTheDocument();
  });

  it('should render the heading with correct text', () => {
    renderWithRouter(<NotFoundPage />);
    const heading = screen.getByRole('heading');
    expect(heading).toHaveTextContent('404 - Page Not Found');
  });

  it('should match snapshot', () => {
    const { asFragment } = renderWithRouter(<NotFoundPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
