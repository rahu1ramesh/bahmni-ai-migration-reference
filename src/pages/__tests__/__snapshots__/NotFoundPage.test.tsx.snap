// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Component should match snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="cds--css-grid"
    >
      <div
        class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-16 cds--css-grid-column"
      >
        <h2>
          404 - Page Not Found
        </h2>
        <p>
          The page you are looking for does not exist.
        </p>
        <a
          data-discover="true"
          href="/"
        >
          Return to Home
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
